ckan/logic/validators.py

```diff
@@ -831,3 +831,10 @@ def empty_if_not_sysadmin(key, data, errors, context):
         return
 
     empty(key, data, errors, context)
+
+def email_validator(value, context):
+    '''Validate email input '''
+    if value:
+        if not re.match("\A(?P<name>[\w\-_]+)@(?P<domain>[\w\-_]+).(?P<toplevel>[\w]+)\Z",value,re.IGNORECASE):
```

Comment:
This is the one we're using, based on the HTM5 spec for email validation https://github.com/open-data/ckanext-canada/blob/master/ckanext/canada/validators.py#L112-L125